src/agent/include/register.hpp

```diff
@@ -1,17 +1,30 @@
 #pragma once
 
-#include <ihttp_client.hpp>
+#include <agent_info.hpp>
+#include <configuration_parser.hpp>
 
 #include <string>
 
-namespace registration
+namespace agent_registration
 {
     struct UserCredentials
     {
         std::string user;
         std::string password;
     };
 
-    bool RegisterAgent(const UserCredentials& userCredentials, http_client::IHttpClient& httpClient);
+    class AgentRegistration
+    {
+    public:
+        AgentRegistration(std::string user, std::string password, const std::string& key, const std::string& name);
+        bool SendRegistration();
```

Comment:
```suggestion
        bool Register();
```
I think this is a better name.

Reply:
I was trying to avoid "Register" because of the keyword "register". How about "RegisterAgent()"?

Reply:
I think there is no problem if the first letter is upper case.

Reply:
Done